### ISSUE TITLE
`Linking.parse` relies on port number which is lost on iOS

### ISSUE THREAD

**Issue by id-ilych** [expo/expo#1721]
### Environment

Environment:
  OS: Windows 7
  Node: 8.11.1
  Yarn: Not Found
  npm: 5.6.0
  Watchman: Not Found
  Xcode: N/A
  Android Studio: Version  3.1.0.0 AI-173.4720617

Packages: (wanted => installed)
  expo: ^27.0.0 => 27.0.1
  react: 16.3.1 => 16.3.1
  react-native: 0.55.2 => 0.55.2

Target:
Tunnelled to Expo app on iOS (development)

### Steps to Reproduce

1. Use `Linking.makeUrl` (e.g. `Linking.makeUrl('script', {id})`)
2. Open that link in Safari (or using QR scaner or whatever)
3. Parse it with `Linking.parse`

### Expected Behavior

`path` should be `script` for the example above

### Actual Behavior

`path` is `##-##.id-ilych.{PROJECT_NAME}.exp.direct/--/script`

#### !!!IMPORTANT!!! 
The bug is not reproducible in Snack because it produces link like
`exp://expo.io/@snack/sdk.27.0.0-#########/--/script?id=123`
Whereas for regular hosted app in tunneling mode it would be like (note port number `:80`)
`##-##.id-ilych.{PROJECT_NAME}.exp.direct:80/--/script?id=123`

The problem is that this `:80` suffix is lost when you trying to open the link from camera (QR) or safari.
`Linking.getInitialURL` gives `##-##.id-ilych.{PROJECT_NAME}.exp.direct/--/script?id=123`
Which makes `Linking.parse` failing to strip off the host uri because it's not a full match now

I do understand that it's probably Safari/Camera/iOS to blame but still that is the environment Expo app is expected to work in so I think it should somehow handle the case.

#### !!!IMPORTANT!!! 

```
import React, { Component } from 'react';
import { Text, View, StyleSheet } from 'react-native';
import { Linking, Constants } from 'expo';

export default class App extends Component {
  
  state = {
    url: 'n/a',
    path: 'n/a',
    queryParams: {},
  }
  
  componentDidMount() {
    Linking.getInitialURL()
      .then((url) => {
        const {path, queryParams} = Linking.parse(url);
        this.setState({url, path, queryParams});
      });
  }

  
  render() {
    var id = 123;
    var url = Linking.makeUrl('script', {id});
    var {path, queryParams} = Linking.parse(url);
    return (
      <View style={styles.container}>
        <Text style={styles.paragraph}>
          Url: {url}
        </Text>
        <Text style={styles.paragraph}>
          Path: {path}
        </Text>
        <Text style={styles.paragraph}>
          QueryParams: {JSON.stringify(queryParams)}
        </Text>
        <Text style={styles.paragraph}>
          Initial URL:
        </Text>
        <Text style={styles.paragraph}>
          Url: {this.state.url}
        </Text>
        <Text style={styles.paragraph}>
          Path: {this.state.path}
        </Text>
        <Text style={styles.paragraph}>
          QueryParams: {JSON.stringify(this.state.queryParams)}
        </Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    paddingTop: Constants.statusBarHeight,
    backgroundColor: '#ecf0f1',
  },
  paragraph: {
    margin: 24,
    fontSize: 18,
    fontWeight: 'bold',
    textAlign: 'center',
    color: '#34495e',
  },
});
```

**Comment by esamelson** [comment]
Hi @id-ilych - thanks for the detailed report. We'll have a fix out for this soon.

**Comment by esamelson** [comment]
Hi all -- sorry for the slow followup here. This has been fixed in `expo@27.0.2` which you can now install via npm/yarn.
